packages/kolibri-common/components/MeteredConnectionNotificationModal.vue

```diff
@@ -28,26 +32,106 @@
 
 <script>
 
+  import { mapGetters } from 'vuex';
+  import urls from 'kolibri.urls';
+  import client from 'kolibri.client';
   import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import { checkCapability } from 'kolibri.utils.appCapabilities';
 
   const Options = Object.freeze({
     DO_NOT_USE_METERED: 'DO_NOT_USE_METERED',
     USE_METERED: 'USE_METERED',
   });
 
+  const meteredNetworkModalDismissedKey = 'METERED_NETWORK_MODAL_DISMISSED';
+
   export default {
     name: 'MeteredConnectionNotificationModal',
     mixins: [commonCoreStrings],
-
     data() {
       return {
         Options,
         selected: Options.DO_NOT_USE_METERED,
+        loading: false,
+        extra_settings: {},
+        dismissed: Boolean(window.sessionStorage.getItem(meteredNetworkModalDismissedKey)),
       };
     },
     computed: {
+      ...mapGetters(['isSuperuser', 'activeConnectionIsMetered']),
       displayMeteredConnectionWarning() {
-        return true;
+        return (
+          this.originalSettingDisallows &&
+          this.activeConnectionIsMetered &&
+          this.isSuperuser &&
+          !this.dismissed
+        );
+      },
+      settingsUrl() {
+        return urls['kolibri:core:devicesettings']();
+      },
+      originalSettingDisallows() {
+        return !this.extra_settings.allow_download_on_metered_connection;
+      },
+    },
+    mounted() {
+      // If we have the ability to check if the device is metered, then
+      // set up a poller to check every 5 seconds and set things up,
+      // otherwise this component will never be rendered anyway so no need to make the calls
+      if (checkCapability('check_is_metered')) {
+        // Get status initially and poll every 5 seconds
+        this.$store.dispatch('updateMeteredStatus').then(() => {
```

Comment:
As you suggest in the question in your PR description, I don't think we should be polling this - only triggering the modal when the user initiates an action that might lead to a download.